lib/charms/tls_certificates_interface/v1/tls_certificates.py

```diff
@@ -1311,6 +1346,22 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
                         chain=certificate["chain"],
                     )
 
+    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
+        """Handler triggered on relation broken event.
+
+        Args:
+            event: Juju event
+
+        Returns:
+            None
+        """
+        relation = self.model.get_relation(self.relationship_name)
+        if not relation:
+            return
+        if not relation.data[relation.app].get("certificates"):  # type: ignore[index]
+            return
+        self.on.all_certificates_invalidated.emit()
```

Comment:
I'm not sure about the `.get("certificates")` guard:
Could there be a use case where lib users would legitimately want to see this event on relation broken, even if there are no certs?